src/NServiceBus.Core/Performance/TimeToBeReceived/ApplyTimeToBeReceivedBehavior.cs

```diff
@@ -17,7 +17,7 @@ public Task Invoke(IOutgoingLogicalMessageContext context, Func<IOutgoingLogical
         {
             if (timeToBeReceivedMappings.TryGetTimeToBeReceived(context.Message.MessageType, out var timeToBeReceived))
             {
-                context.Extensions.AddDeliveryConstraint(new DiscardIfNotReceivedBefore(timeToBeReceived));
+                context.Extensions.Get<OperationProperties>().DiscardIfNotReceivedBefore = new DiscardIfNotReceivedBefore(timeToBeReceived);
```

Comment:
do we guarantee that OperationProperties is set or use GetOrCreate instead? Should we have a test for that?